src/Common/src/TypeSystem/CodeGen/TargetDetails.CodeGen.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.TypeSystem
+{
+    // Extension to TargetDetails related to code generation
```

Comment:
We might want to kick things like the `Abi` to this flavor of `TargetDetails` too.